torch/testing/_internal/common_utils.py

```diff
@@ -807,10 +861,817 @@ def wrapper(*args, **kwargs):
+    torch._pack_padded_sequence,
+    torch._pad_packed_sequence,
+    torch.sparse_coo_tensor,
+    # RuntimeError: Expected 3D or 4D (batch mode) tensor with optional 0 dim
+    # batch size for input, but got:[1, 1, 0]
+    # in test_nn.py TestNNDeviceTypeCPU.test_max_pool1d_corner_cases_cpu_float64
+    torch.nn.functional.max_pool1d,
+    # IndexError: select() cannot be applied to a 0-dim tensor.
+    # e.g. test_fn_fwgrad_bwgrad_index_add_cpu_complex128 (__main__.TestGradientsCPU)
+    torch.index_add,
+    torch.Tensor.index_add,
+    torch.Tensor.index_add_,
+    # Can't copy out of meta tensor
+    torch.linalg.eigvals,
+    torch.linalg.lu_factor,
+    torch.nn.functional.ctc_loss,
+    # Our conversion to meta is not accurate enough (doesn't
+    # preserve storage_offset, e.g.)
+    torch.Tensor.as_strided,
+    # This one segfaults when you call it
+    torch.Tensor.type,
+    # We don't clone autograd history, so this will generally not work
+    torch.autograd.grad,
+    torch.Tensor.backward,
+    torch.Tensor.__deepcopy__,
+    # Don't do factories
+    torch.ones,
+    torch.full,
+    torch.empty,
+    torch.randperm,
+    torch.logspace,
+    torch.zeros,
+    torch.arange,
+    torch.vander,
+    torch.as_tensor,
+    torch.tensor,
+    torch.zeros_like,
+    torch.randn_like,
+    torch.sparse_csr_tensor,
+    torch._sparse_coo_tensor_unsafe,
+    torch._sparse_csr_tensor_unsafe,
+    torch._validate_sparse_csr_tensor_args,
+}
+
+skipped = set()
+
+RE_NOT_IMPLEMENTED_MSG = re.compile(r"Could not run '([^']+)' with arguments ")
+
+class CoverageMode(torch.overrides.TorchFunctionMode):
+    if hasattr(__main__, '__file__'):
+        prefix = Path(__file__).parent.parent.parent.parent
+        full_test_file = __main__.__file__
+        test_file = os.path.relpath(full_test_file, prefix)
+    else:
+        test_file = '<stdin>'
+
+    def __init__(self, conn, test_case):
+        self.conn = conn
+        self.test_case = test_case
+        self.seen = set()
+
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        self.seen.add(func)
+        if kwargs is None:
+            kwargs = {}
+        return func(*args, **kwargs)
+
+    def commit(self):
+        rows = [
+            (
+                self.test_file,
+                self.test_case.id(),
+                torch.overrides.resolve_name(func) or str(func)
+            )
+            for func in self.seen
+        ]
+        self.conn.executemany("INSERT INTO calls_view VALUES (?,?,?)", rows)
+        self.conn.commit()
+
+    def __del__(self):
+        self.commit()
+
 class CrossRefMode(torch.overrides.TorchFunctionMode):
+    def __init__(self, test_case):
+        self.test_case = test_case
+
     def __torch_function__(self, func, types, args=(), kwargs=None):
         kwargs = kwargs or {}
+
+        hit = 0
+        miss = 0
+
+        # Doesn't actually return a storage
+        @functools.lru_cache(None)
```

Comment:
We call this a lot with the same Storage?

Reply:
not really, but it's important to give the same tensor back if you hit the same storage